wizeline/project-lab
# Feature/slack interactions - #97 - adds the ability to search for projects and like them through Slack

**What does this PR do?**

It adds base functionality for interacting with Slack including sending a notification to a channel when a new proposal is created and searching for proposals/projects and liking them

**Where should the reviewer start?**

app/pages/projects/create/index.tsx 

and
 
app/api/slackEvents.tsx

**How should this be manually tested?**

You should first subscribe to the Slack workspace created by @shamcalderon here:
https://join.slack.com/t/testapi-bqp3422/shared_invite/zt-x4zx66qv-Sl0bYx9WAIazlURs3~TLZQ

Then, you need to update the app's config with the path to your project, e.g. if you're running on your local machine, you can use [ngrok](https://ngrok.com/) to get an URL.

The path to slackEvents goes here: https://api.slack.com/apps/A02HSR9RKCN/event-subscriptions? 
and the path to slackInteractions goes here: https://api.slack.com/apps/A02HSR9RKCN/interactive-messages?

You should also add these two variables to your .env.local. (maybe we could add the channel as an input later on...)
SLACK_TOKEN=Jcrmd62XVVFDUQ8RDoG4aa65
SLACK_CHANNEL=C02JP7BP6TA 
APP_TOKEN = xoxb-2611588638355-2611599839219-fwXj030fUtAAV7U1jN3Ic527

We're using #api-project for the tests so when you create a new proposal, a card with the information should appear there. You can also call the app with @ and ask for information about a project using this format:
@[app name] Info name/search [project-name]
e.g. 

```
@Project Lab - Test Info name WizePace
or @Project Lab - Test Info search wize
```


**Any background context you want to provide?**

I changed the structure of the async functions a little, based on suggestions from the team. When you use the search "functionality" I'm using Promise.all to try and add info about the project's owner to the cards for each project. As I understand, promises in an array are resolved in order by Promise.all so I tried nesting two of those. 

I had never coded something like that so it might need a little work. Feedback on that part ( as well as the rest of it ) would be appreciated.


**What are the relevant tickets?**

[#97](https://github.com/wizeline/project-lab/issues/97)
